snuba/consumers/consumer_builder.py

```diff
@@ -109,29 +129,42 @@ def __init__(
 
         stream_loader = self.storage.get_table_writer().get_stream_loader()
 
-        self.raw_topic: Topic
+        self.raw_topic: ArroyoTopic
         if kafka_params.raw_topic is not None:
-            self.raw_topic = Topic(kafka_params.raw_topic)
+            # we do not support topic override in a sliced context
+            assert slice_id is None
+            self.raw_topic = ArroyoTopic(kafka_params.raw_topic)
         else:
-            self.raw_topic = Topic(stream_loader.get_default_topic_spec().topic_name)
+            self.raw_topic = ArroyoTopic(
+                stream_loader.get_default_topic_spec().topic_name
+            )
 
-        self.replacements_topic: Optional[Topic]
+        self.replacements_topic: Optional[ArroyoTopic]
         if kafka_params.replacements_topic is not None:
-            self.replacements_topic = Topic(kafka_params.replacements_topic)
+            self.replacements_topic = ArroyoTopic(kafka_params.replacements_topic)
         else:
             replacement_topic_spec = stream_loader.get_replacement_topic_spec()
             if replacement_topic_spec is not None:
-                self.replacements_topic = Topic(replacement_topic_spec.topic_name)
+                self.replacements_topic = ArroyoTopic(replacement_topic_spec.topic_name)
             else:
                 self.replacements_topic = None
 
-        self.commit_log_topic: Optional[Topic]
+        self.commit_log_topic: Optional[ArroyoTopic]
+
         if kafka_params.commit_log_topic is not None:
-            self.commit_log_topic = Topic(kafka_params.commit_log_topic)
+            kafka_param_topic_spec = KafkaTopicSpec(
+                SnubaTopic(kafka_params.commit_log_topic)
+            )
+            self.commit_log_topic = ArroyoTopic(
+                kafka_param_topic_spec.get_physical_topic_name(slice_id)
```

Comment:
You should not be calling get_physical_topic_name here since we were already given the actual topic name that needs no further resolving. We only want it below in the else statement.